LibreNMS/Modules/Ipv4Addresses.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * Ipv4Address.php
+ *
+ * Ipv4 addresses discovery module
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @link       http://librenms.org
+ *
+ * @copyright  2025 Peca Nesovanovic
+ * @author     Peca Nesovanovic <peca.nesovanovic@sattrakt.com>
+ */
+
+namespace LibreNMS\Modules;
+
+use App\Facades\PortCache;
+use App\Models\Device;
+use App\Models\Ipv4Address;
+use App\Models\Ipv4Network;
+use App\Observers\ModuleModelObserver;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\DB\SyncsModels;
+use LibreNMS\Exceptions\InvalidIpException;
+use LibreNMS\Interfaces\Data\DataStorageInterface;
+use LibreNMS\Interfaces\Discovery\Ipv4AddressDiscovery;
+use LibreNMS\Interfaces\Module;
+use LibreNMS\OS;
+use LibreNMS\Polling\ModuleStatus;
+use LibreNMS\Util\IPv4;
+use SnmpQuery;
+
+class Ipv4Addresses implements Module
+{
+    use SyncsModels;
+
+    /**
+     * @inheritDoc
+     */
+    public function dependencies(): array
+    {
+        return ['ports'];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldDiscover(OS $os, ModuleStatus $status): bool
+    {
+        return $status->isEnabledAndDeviceUp($os->getDevice());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function shouldPoll(OS $os, ModuleStatus $status): bool
+    {
+        return false;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function discover(OS $os): void
+    {
+        $ips = new Collection;
+        if ($os instanceof Ipv4AddressDiscovery) {
+            $ips = $os->discoverIpv4Addresses();
+        }
+        if ($ips->isEmpty()) {
+            $ips = $this->discoverIpMib($os->getDevice());
+        }
+
+        $ips = $ips->filter(function ($data) {
+            $addr = trim(str_replace('"', '', $data->ipv4_address ?? ''));
+            $context = trim(str_replace('"', '', $data->context_name ?? ''));
+            $prefix = trim($data->ipv4_prefixlen ?? '');
+
+            if ($addr == '' || $prefix == '') {
```

Comment:
Hi @Npeca75 
You could probably use the default classfull here whenever prefix is not provided.
In fact, you could probably implement some kind of helper function that does this compute, and call it from here as well as from the Route module. 

```
                $tmp = explode('.', $dst);
                if ($tmp[0] < 128) {
                    $prefix = '255.0.0.0';
                }
                if ($tmp[0] > 128 && $tmp[0] < 192) {
                    $prefix = '255.255.0.0';
                }
                if ($tmp[0] > 192 && $tmp[0] < 224) {
                    $prefix = '255.255.255.0';
                }
                Log::info('Classfull netmask from RFC: ' . $dst . ' - ' . $hop . ' - ' . $prefix);
```